languagetool-core/src/main/java/org/languagetool/CheckResults.java

```diff
@@ -18,45 +18,49 @@
  */
 package org.languagetool;
 
+import lombok.Getter;
 import org.jetbrains.annotations.NotNull;
 import org.languagetool.rules.RuleMatch;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Objects;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * @since 5.3
  */
 public class CheckResults {
 
+  @Getter
   private List<RuleMatch> ruleMatches;
-  private List<Range> ignoredRanges;
+  @Getter
+  private final List<Range> ignoredRanges;
+  private final List<ExtendedSentenceRange> extendedSentenceRanges;
   private final List<SentenceRange> sentenceRanges = new ArrayList<>();
 
-  public CheckResults(List<RuleMatch> ruleMatches, List<Range> ignoredRanges) {
-    this.ruleMatches = Objects.requireNonNull(ruleMatches);
-    this.ignoredRanges = Objects.requireNonNull(ignoredRanges);
-  }
 
-  public List<Range> getIgnoredRanges() {
-    return ignoredRanges;
+  public CheckResults(List<RuleMatch> ruleMatches, List<Range> ignoredRanges) {
+    this(ruleMatches, ignoredRanges, Collections.emptyList());
   }
 
-  public List<RuleMatch> getRuleMatches() {
-    return ruleMatches;
+  public CheckResults(List<RuleMatch> ruleMatches, List<Range> ignoredRanges, List<ExtendedSentenceRange> extendedSentenceRanges) {
+    this.ruleMatches = Objects.requireNonNull(ruleMatches);
+    this.ignoredRanges = Objects.requireNonNull(ignoredRanges);
+    this.extendedSentenceRanges = Objects.requireNonNull(extendedSentenceRanges);
+    //TODO: use this later
```

Comment:
what about this TODO? can you add more information, implement it now or delete it?